pkg/controller/ignition/controller.go

```diff
@@ -0,0 +1,216 @@
+				"Service", naming.ManualRef(c.namespace, c.serviceName),
+			)
+			ignited = false
+		} else {
+			klog.V(2).InfoS(
+				"Service is available and has an IP address",
+				"Service", naming.ManualRef(svc.Namespace, svc.Name),
+				"UID", svc.UID,
+			)
+		}
+	}
+
+	pod, err := c.podLister.Pods(c.namespace).Get(c.serviceName)
+	if err != nil {
+		return fmt.Errorf("can't get pod %q: %w", c.serviceName, err)
+	}
+
+	if len(pod.Status.PodIP) == 0 {
+		klog.V(2).InfoS("PodIP is not yet set", "Pod", klog.KObj(pod), "UID", pod.UID)
+		ignited = false
+	} else {
+		klog.V(2).InfoS("PodIP is present on the Pod", "Pod", klog.KObj(pod), "UID", pod.UID, "IP", pod.Status.PodIP)
+	}
+
+	containerID, err := controllerhelpers.GetScyllaContainerID(pod)
+	if err != nil {
+		return controllertools.NonRetriable(
+			fmt.Errorf("can't get scylla container id in pod %q: %v", naming.ObjRef(pod), err),
+		)
+	}
+
+	if len(containerID) == 0 {
+		klog.V(2).InfoS("ContainerID is not yet set", "Pod", klog.KObj(pod), "UID", pod.UID)
+		ignited = false
+	} else {
+		klog.V(2).InfoS("Pod has ContainerID set", "Pod", klog.KObj(pod), "UID", pod.UID, "ContainerID", containerID)
+	}
+
+	cmLabelSelector := labels.Set{
+		naming.OwnerUIDLabel:      string(pod.UID),
+		naming.ConfigMapTypeLabel: string(naming.NodeConfigDataConfigMapType),
+	}.AsSelector()
+	configMaps, err := c.configMapLister.ConfigMaps(c.namespace).List(cmLabelSelector)
+	if err != nil {
+		return fmt.Errorf("can't list tuning configmap: %w", err)
+	}
+
+	switch l := len(configMaps); l {
+	case 0:
+		klog.V(2).InfoS("Tuning ConfigMap for pod is not yet available", "Pod", klog.KObj(pod), "UID", pod.UID)
+		ignited = false
+
+	case 1:
+		cm := configMaps[0]
+		src := &internalapi.SidecarRuntimeConfig{}
+		src, err = controllerhelpers.GetSidecarRuntimeConfigFromConfigMap(cm)
+		if err != nil {
+			return controllertools.NonRetriable(
+				fmt.Errorf("can't get sidecar runtime config from configmap %q: %w", naming.ObjRef(cm), err),
+			)
+		}
+
+		if containerID == src.ContainerID {
+			if len(src.BlockingNodeConfigs) > 0 {
+				klog.V(2).InfoS("Waiting on NodeConfig(s)",
+					"ConfigMap", klog.KObj(cm),
+					"ContainerID", containerID,
+					"NodeConfig", src.BlockingNodeConfigs,
+				)
+				ignited = false
+			}
+		} else {
+			klog.V(2).InfoS("Scylla runtime config is not yet updated with our ContainerID",
+				"ConfigMap", klog.KObj(cm),
+				"ConfigContainerID", src.ContainerID,
+				"SidecarContainerID", containerID,
+			)
+			ignited = false
+		}
+
+	default:
+		return fmt.Errorf("mutiple tuning configmaps are present for pod %q with UID %q", naming.ObjRef(pod), pod.UID)
+	}
+
+	if ignited {
+		klog.V(2).InfoS("Ignition successful", "SignalFile", naming.ScyllaDBIgnitionDonePath)
+	} else {
+		klog.V(2).InfoS("Waiting for ignition to complete.", "SignalFile", naming.ScyllaDBIgnitionDonePath)
+	}
+
+	klog.V(2).InfoS("Updating ignition", "Ignited", ignited, "SignalFile", naming.ScyllaDBIgnitionDonePath)
```

Comment:
nit: Updating ignition state

Reply:
Not state per same, the state can be identical mostly, it touches just the timestamp in most cases. There is an independent log regarding the state.